rbd.c

```diff
@@ -943,8 +943,18 @@ static void rbd_finish_aio_generic(rbd_completion_t completion,
 {
 	struct tcmu_device *dev = aio_cb->dev;
 	struct tcmulib_cmd *tcmulib_cmd = aio_cb->tcmulib_cmd;
-	struct iovec *iovec = tcmulib_cmd->iovec;
-	size_t iov_cnt = tcmulib_cmd->iov_cnt;
+
+    struct iovec *iovec;
+    size_t iov_cnt;
+    if ((tcmulib_cmd->cdb[0] & 0x83) == 0x83){
```

Comment:
Each handler should should not have to know what command is being executed and know this low level of details about info like cmdstate.

We have wanted to rewrite the iovec code, because some handlers modify the iovec and some do not. The tcmur_cmd_handler code was supposed to temporarily hack around this and handle it for all handlers. Let me get back to you on this in a day or so when I have more time. I am on PTO right now, so do not have access to everything I would like to be able to give you a good review comment on how to handle this properly.